### PR TITLE
schemas/0.1.0: add minLength to strings to avoid empty strings; add missing examples

### PR DESCRIPTION
**Pull request by ndushay**
add a minLength to string properties/attributes to avoid empty strings in actual templates.

This was prompted by https://github.com/LD4P/sinopia_editor/pull/318 and https://github.com/LD4P/sinopia_editor/issues/315, as well as https://github.com/LD4P/sinopia_profile_editor/issues/178

I also added examples, where they were missing.

FYI:  I haven't found any profile level remarks ... perhaps that can go away???

### CODE REVIEW

File: schemas/0.1.0/resource-template.json [hunk 1]
```diff
@@ -10,8 +10,8 @@
     "date",
     "id",
     "propertyTemplates",
-    "resourceURI",
     "resourceLabel",
+    "resourceURI",
```
Comment (ndushay) [inline schemas/0.1.0/resource-template.json]: out of alpha order!  Shame on me!

File: schemas/0.1.0/profile.json [hunk 2]
```diff
@@ -71,31 +76,41 @@
...
+          ]
         },
         "source": {
           "type": "string",
           "format": "uri",
           "title": "Source URI",
           "description": "URL or URI for the profile in the author's source repository or space.",
+          "minLength": 5,
           "example": [
-            "https://raw.githubusercontent.com/LD4P/sinopia_sample_profiles/master/verso/BIBFRAME%202.0%20Admin%20Metadata.json"
```
Comment (ndushay) [inline schemas/0.1.0/profile.json]: Admin Metadata is not a great example; it's too atypical.

File: schemas/0.1.0/property-template.json [hunk 3]
```diff
@@ -77,10 +85,10 @@
           "items": {
             "type": "string"
           },
+          "minItems": 1,
           "example": [
-            "['profile:bf2:Agent:Person', 'profile:bf2:Identifiers:Barcode', 'profile:bflc:Agents:PrimaryContribution']"
-          ],
-          "minItems": 1
+            "['sinopia:resourceTemplate:bf2:Agent:Person', 'sinopia:resourceTemplate:bf2:Identifiers:Barcode', 'sinopia:resourceTemplate:bflc:Agents:PrimaryContribution']"
+          ]
```
Comment (ndushay) [inline schemas/0.1.0/property-template.json]: I figure better to refer to less confusing resource template ids!

File: schemas/0.1.0/property-template.json [hunk 4]
```diff
@@ -89,10 +97,10 @@
           "items": {
             "type": "string"
           },
+          "minItems": 1,
```
Comment (ndushay) [inline schemas/0.1.0/property-template.json]: moved this up to be more consistent

**Comment by jermnelson** [comment]
With these changes, should the Patch version be incremented? (i.e. 0.1.1)

File: schemas/0.1.0/profile.json [hunk 5]
```diff
@@ -71,31 +76,41 @@
         "remark": {
           "type": "string",
           "title": "Remark",
-          "description": "Comment or guiding statement intended to be presented as supplementary information in user display."
+          "description": "Comment or guiding statement intended to be presented as supplementary information in user display.",
+          "minLength": 3,
```
Comment (jermnelson) [inline schemas/0.1.0/profile.json]: By making this a minLength of 3, does this mean that the `remark` is now required for validation?

**Comment by ndushay** [comment]
@jermnelson yes, in theory the patch version would be incremented but:

- when the first number is 0, anything goes
- there were no changes needed to our actual profiles/templates

So rather than have a confusing proliferation of schema versions at the outset, I thought it would be easier to just keep the version.   If a schema change in the 0 range *does* require profile changes, then definitely kick me to make a new schema version and update the profiles/templates accordingly.

File: schemas/0.1.0/property-template.json (hunk 3 above)
Comment (jermnelson) [reply schemas/0.1.0/property-template.json]: :+1:

File: schemas/0.1.0/profile.json (hunk 5 above)
Comment (ndushay) [reply schemas/0.1.0/profile.json]: @jermnelson no - it means when it is _PRESENT_ it must have a minimum length of 3.  required would show up around line 15 - in the list of required properties.  :-)

**Pull request merged**